File: schemas/token.py
from uuid import UUID
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional


class CreateTokenRequest(BaseModel):
    name: str
    expires_at: Optional[datetime] = None


class UpdateTokenNameRequest(BaseModel):
    name: str


class UpdateTokenStatusRequest(BaseModel):
    is_active: bool


class TokenResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    org_id: UUID
    name: str
    is_active: bool
    created_by: UUID
    expires_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

File: frontend/app/types/token.ts
export interface AuthTokenResponse {
  id: string
  org_id: string
  name: string
  created_by: string
  is_active: boolean
  expires_at: string | null
  created_at: string
  updated_at: string
}

export interface CreateTokenResponse {
  name: string
  token: string
  message: string
}

export interface AuthToken {
  id: string
  orgId: string
  name: string
  token?: string
  createdBy: string
  isActive: boolean
  expiresAt: string | null
  createdAt: string
  updatedAt: string
}
